include/effectmanager.h

```diff
@@ -527,11 +527,7 @@ class  EffectManager : public IJSONSerializable
 
     const bool AreEffectsEnabled() const
     {
-        for (auto& pEffect : _vEffects)
-            if (pEffect->IsEnabled())
-                return true;
-
-        return false;
+        return std::any_of(_vEffects.begin(), _vEffects.end(), [](const auto& pEffect){ return pEffect->IsEnabled(); } );

```

Comment:
OK, maybe this is a bit too clever, but it's short!

Reply:
Optimizers love std::algorithm code like this because it completely explains the intent. Vectorizers munch on that kind of code.

It took me a while to warm up to it, too.

Reply:
The name of the function itself does. I just find that the semantics of the specification of the iterator begin and end, and the ratio of the lambda fluff around the _actual_ "thing" (is it enabled or not) distracting. But I already stated I'm happy to accept this as a matter of personal preference.

Maybe the underlying question is what level of comfort with this form of C++ we expect people to have if they want to contribute to the core of the project. I'm definitely not going to start that discussion, though. :)